plugin/cog/cog-rangereader-azure/src/main/java/it/geosolutions/imageioimpl/plugins/cog/AzureConfigurationProperties.java

```diff
@@ -52,47 +51,46 @@ public class AzureConfigurationProperties {
 
         if (path.startsWith("https")) {
             // replace for quicker finding
-            path = path.replace("https", "http");
+            path = path.replaceFirst("https", "http");
             useHTTPS = true;
         }
 
-        BlobURLParts parts = null;
         try {
-            parts = URLParser.parse(uri.toURL());
-            container = parts.containerName();
-            String blobName = parts.blobName();
+            BlobUrlParts parts = BlobUrlParts.parse(uri.toURL());
+            container = parts.getBlobContainerName();
+            String blobName = parts.getBlobName();
             int lastIndex = blobName.lastIndexOf("/");
             if (lastIndex > 0) {
                 prefix = blobName.substring(0, lastIndex);
             }
-            String host = parts.host();
+            String host = parts.getHost();
             int blobcoreIdx = host.indexOf("." + AzureClient.AZURE_URL_BASE);
             if (blobcoreIdx > 0) {
                 accountName = host.substring(0, blobcoreIdx);
             }
 
-        } catch (UnknownHostException| MalformedURLException e) {
+        } catch (IllegalStateException| MalformedURLException e) {
             throw new RuntimeException("Unable to parse the provided uri " + path + "due to " + e.getLocalizedMessage());
         }
 
-        if (container == null) {
+        if (container == null) {//REVISIT: dead code
```

Comment:
Why "dead code"? Because the code above will never return a null container?
Hum... looks like that's the case indeed: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobUrlParts.java#L423

Reply:
yup, I tried to make a test case for those code paths but couldn't

Reply:
oh, also I think the values should be obtained with ``PropertyLocator.getPropertyValue()` instead of `PropertyLocator.getEnvironmentValue()` cause some keys use mixed-case (`azure.reader.accountName`, `azure.reader.accountKey`) and `getEnvironmentValue()` will try the system property with `toLowerCase()`, but I'm not sure if it's already being used as-is